src/js/utils/merge-options.js

```diff
@@ -11,38 +11,50 @@ function isPlain(obj) {
 }
 
 /**
- * Merge two options objects, recursively merging **only* * plain object
- * properties.  Previously `deepMerge`.
+ * Merge customizer. video.js simply overwrites non-simple objects
+ * (like arrays) instead of attempting to overlay them.
+ * @see https://lodash.com/docs#merge
+ */
+const customizer = function(destination, source) {
+  // If we're not working with a plain object, copy the value as is
+  // If source is an array, for instance, it will replace destination
+  if (!isPlain(source)) {
+    return source;
+  }
+
+  // If the new value is a plain object but the first object value is not
+  // we need to create a new object for the first object to merge with.
+  // This makes it consistent with how merge() works by default
+  // and also protects from later changes the to first object affecting
+  // the second object's values.
+  if (!isPlain(destination)) {
+    return mergeOptions({}, source);
+  }
+};
+
+/**
+ * Merge one or more options objects, recursively merging **only**
+ * plain object properties.  Previously `deepMerge`.
  *
- * @param  {Object} object    The destination object
- * @param  {...Object} source One or more objects to merge into the first
- * @returns {Object}          The updated first object
+ * @param  {...Object} source One or more objects to merge
+ * @returns {Object}          a new object that is the union of all
+ * provided objects
  * @function mergeOptions
  */
-export default function mergeOptions(object={}) {
-
-  // Allow for infinite additional object args to merge
-  Array.prototype.slice.call(arguments, 1).forEach(function(source){
-
-    // Recursively merge only plain objects
-    // All other values will be directly copied
-    merge(object, source, function(a, b) {
-
-      // If we're not working with a plain object, copy the value as is
-      if (!isPlain(b)) {
-        return b;
-      }
-
-      // If the new value is a plain object but the first object value is not
-      // we need to create a new object for the first object to merge with.
-      // This makes it consistent with how merge() works by default
-      // and also protects from later changes the to first object affecting
-      // the second object's values.
-      if (!isPlain(a)) {
-        return mergeOptions({}, b);
-      }
-    });
-  });
-
-  return object;
+export default function mergeOptions() {
```

Comment:
The only changes in this file are related to making mergeOptions() non-mutating.